src/macros.jl

```diff
@@ -219,10 +220,19 @@ If a `@testitem` passes on retry, then it will be recorded as passing in the tes
     @testitem "Flaky test" retries=1 begin
         @test rand() < 1e-4
     end
+
+If a `@testitem` should be aborted after a certain period of time, e.g. the test is known
+to occassionally hang, then you can set a timeout (in seconds) by passing the `timeout` keyword.
+Note that `timeout` currently only works when tests are run with multiple workers.
```

Comment:
why is that? (out of curiosity)

Reply:
I think to robustly cancel a task on the same process, we'd need something like https://github.com/JuliaLang/julia/issues/6283